File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void temp(TreeNode* node, unordered_map<int, int>& mp, int depth){
        if(!node) return;
        cout << node->val << endl;
        if(mp.find(depth) == mp.end()) 
            mp[depth] = node->val;
        temp(node->right, mp, depth+1);
        temp(node->left, mp, depth+1);
    }

    vector<int> rightSideView(TreeNode* root) {
        vector<int> res;
        if(!root) return res;
        unordered_map<int, int> mp;
        temp(root, mp, 0);
        for(int i=0;i<mp.size();i++){
            res.push_back(mp[i]);
        }
        return res;
    }
};

File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.dart
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *   int val;
 *   TreeNode? left;
 *   TreeNode? right;
 *   TreeNode([this.val = 0, this.left, this.right]);
 * }
 */
class Solution {
  List<int> rightSideView(TreeNode? root) {
    List<List<int>> res = [];
    solve(res, root, 0);
    List<int> ans = [];
    for(int i=0;i<res.length;i++){
        ans.add(res[i][res[i].length-1]);
    }
    return ans;
  }
  void solve(List<List<int>> res, TreeNode? root, int index){
    if(root == null) return;
    if(res.length <= index) res.add([]);
    res[index].add(root.val);
    solve(res, root.left, index+1);
    solve(res, root.right, index+1);
  }
}